miri-script/src/commands.rs

```diff
@@ -470,29 +480,26 @@ impl Command {
         // Scan for "--target" to overwrite the "MIRI_TEST_TARGET" env var so
         // that we set the MIRI_SYSROOT up the right way. We must make sure that
         // MIRI_TEST_TARGET and `--target` are in sync.
-        use itertools::Itertools;
-        let target = flags
-            .iter()
-            .take_while(|arg| *arg != "--")
-            .tuple_windows()
-            .find(|(first, _)| *first == "--target");
-        if let Some((_, target)) = target {
-            // Found it!
+        let target = get_target(&flags)?;
+
+        if let Some(target) = target.as_deref() {
+            // If cross-interpreting, we must set the flags accordingly.
             e.sh.set_var("MIRI_TEST_TARGET", target);
-        } else if let Ok(target) = std::env::var("MIRI_TEST_TARGET") {
-            // Convert `MIRI_TEST_TARGET` into `--target`.
-            flags.push("--target".into());
-            flags.push(target.into());
+
+            // Set target if not in the arguments
+            if !has_flag(&flags, "target") {
+                flags.push("--target".into());
+                flags.push(target.into());
+            }
```

Comment:
I don't see the point of this part of the PR.

Reply:
It preserves what it was doing before, keeping that environment variable and CLI args in sync.

Reply:
We already did that before though -- we supported both the env var and the CLI arg. I don't think any change is needed in `fn run`.

Reply:
There is no semantic change. It does what the current HEAD does. In current head, if the enviroment variable was set, but there is not a `--target` flag, it inserts them.

https://github.com/rust-lang/miri/blob/d9188da764dafd6f46affeef03ec8d4f96e9640c/miri-script/src/commands.rs#L482-L486

I can test if it works without keeping them in sync though.

Reply:
> There is no semantic change.

Right, so the code looks more complicated than before and there's no semantic change... so why do this...?

Reply:
Because it was done in multiple places previously so I factored it to its own function. I will revert it.

Reply:
I don't think it was? it was multiple places with your first attempt for `fn test` but we changed that and now it's just `fn run`.

FWIW I am planning a follow-up PR that entirely removes MIRI_TEST_TARGET from the public interface. I never liked having two ways of setting the target for `./miri run`, and now that you're adding `--target` support for `./miri test` we can just use `--target` everywhere.

Reply:
Yeah, I meant in a previous attempt. 

I made the changes for removing MIRI_TEST_TARGET then got to the ci script and realized I don't know bash well enough to do those changes.